pkg/controller/engine.go

```diff
@@ -160,6 +163,14 @@ func For(kind client.Object, h handler.EventHandler, p ...predicate.Predicate) W
 	return Watch{kind: kind, handler: h, predicates: p}
 }
 
+// TriggeredBy returns a custom watch for secondary resources triggering the
+// controller. source.Kind can be used to create a source for a secondary cache.
+// Events will be handled by the supplied EventHandler, and may be filtered by
+// the supplied predicates.
+func TriggeredBy(source source.Source, h handler.EventHandler, p ...predicate.Predicate) Watch {
```

Comment:
@negz see that I slightly changed the signature. It's more flexible now.

Before: `TriggeredBy(someCache, kind, h, p)`
Now: `TriggeredBy(source.Kind(someCache, kind), h, p)`